crates/miden-tx/src/prover/mod.rs

```diff
@@ -121,32 +121,36 @@ impl LocalTransactionProver {
         tx_witness: TransactionWitness,
     ) -> Result<ProvenTransaction, TransactionProverError> {
         let TransactionWitness {
-            tx_inputs,
+            prep_inputs,
+            input_notes,
             tx_args,
             foreign_account_code,
             advice_witness,
         } = tx_witness;
 
-        let (stack_inputs, advice_inputs) =
-            TransactionKernel::prepare_inputs(&tx_inputs, &tx_args, Some(advice_witness))
-                .map_err(TransactionProverError::ConflictingAdviceMapEntry)?;
+        let kernel_inputs =
+            TransactionKernelInputs::new(prep_inputs, input_notes, tx_args, Some(advice_witness))
+                .expect("transaction witness must make valid kernel inputs");
```

Comment:
Assuming we still keep `TransactionWitness`, this could be just:

```rust
let kernel_inputs = tx_witness.into();
```

Reply:
The use case here is ultimately one of `into_parts()` of the witness but we just do that via the pub fields instead of such a fn

Reply:
Removed tx witness in any case